services/src/main/scala/cromwell/services/instrumentation/AsynchronousThrottlingGaugeMetricActor.scala

```diff
@@ -0,0 +1,73 @@
+package cromwell.services.instrumentation
+
+import akka.actor.{ActorRef, LoggingFSM, Props, Status}
+import akka.pattern.pipe
+import cats.data.NonEmptyList
+import cromwell.services.instrumentation.AsynchronousThrottlingGaugeMetricActor._
+
+import scala.concurrent.Future
+
+class AsynchronousThrottlingGaugeMetricActor(metricPath: NonEmptyList[String],
+                                             instrumentationPrefix: Option[String],
+                                             override val serviceRegistryActor: ActorRef)
+  extends LoggingFSM[AsynchronousThrottlingGaugeMetricActorState, Unit]
+    with CromwellInstrumentation {
+
+  implicit val ec = context.dispatcher
+
+  startWith(WaitingForMetricCalculationRequestOrMetricValue, ())
+
+  when(WaitingForMetricCalculationRequestOrMetricValue) {
+    case Event(MetricValue(value), _) =>
+      sendGaugeAndStay(value)
+    case Event(CalculateMetricValue(calculateMetricValueFunction), _) =>
+      calculateMetricValueFunction()
+        .map(intValue => FinishedMetricValueCalculation(intValue.toLong))
+        .pipeTo(self)
+      goto(MetricCalculationInProgress)
+  }
+
+  when(MetricCalculationInProgress) {
+    case Event(CalculateMetricValue, _) =>
+      // metric actor is already busy calculating metric value, so we dismiss this request
+      stay()
+    case Event(MetricValue(value), _) =>
+      // pass through the pre-calculated metric value and stay, continuing to wait for ongoing metric value calculation to finish
+      sendGaugeAndStay(value)
+    case Event(FinishedMetricValueCalculation(value), _) =>
+      sendGauge(metricPath, value, instrumentationPrefix)
+      goto(WaitingForMetricCalculationRequestOrMetricValue)
+    case Event(Status.Failure(ex), _) =>
+      log.error(s"Cannot send gauge metric value: error occurred while evaluating Future value.", ex)
+      goto(WaitingForMetricCalculationRequestOrMetricValue)
+  }
+
+  whenUnhandled {
+    case Event(unexpected, _) =>
+      log.warning(s"Programmer error: this actor should not receive message $unexpected while in state $stateName")
```

Comment:
You can also put in a `from ${sender.path}` to add a little more debugging help if this ever happens.